File: benchmarks/pb_total_list/mono.php
<?php

function f_rcv($ary) {
	$ct = 0;
	foreach($ary as $e) {
		$ct += $e;
	}
	return $ct;
}

function f_call($n) {

	$a = array();
	for ($i = 0; $i < 1000; $i++) {
		$a[] = $i;
	}

	$v = 0;
	for ($i = 0; $i < $n; $i++) {
		$v += f_rcv($a);
	}

	$expect = 499500 * $n;
	assert($v == $expect);
}

function run_iter($n) {
	f_call($n);
}

?>

File: benchmarks/pb_total_list/mono.py
def f_rcv(ary):
    ct = 0
    for i in ary:
        ct += i
    return ct

def f_call(n):
    a = range(1000)

    v = 0
    i = 0
    while i < n:
        v += f_rcv(a)
        i += 1

    expect = 499500 * n
    assert v == expect

def run_iter(n):
    f_call(n)
